dll/cpl/intl/languages.c

```diff
@@ -1,15 +1,31 @@
 #include "intl.h"
 
 #include <shellapi.h>
+#include <shlobj.h>
 #include <strsafe.h>
 
-/* Is there any Japanese input method? */
-BOOL HasJapaneseIME(VOID)
+/* Where is the uninstaller of "ReactOS JPN Package"? */
+BOOL GetJapaneseUninstallPath(HWND hwnd, LPWSTR pszPath, SIZE_T cchPath)
 {
-    WCHAR szImePath[MAX_PATH];
-    GetSystemDirectoryW(szImePath, _countof(szImePath));
-    StringCchCatW(szImePath, _countof(szImePath), L"\\mzimeja.ime");
-    return GetFileAttributesW(szImePath) != INVALID_FILE_ATTRIBUTES;
+    SHGetSpecialFolderPathW(hwnd, pszPath, CSIDL_PROGRAM_FILES, FALSE);
```

Comment:
Isn't this information stored in the registry already? (by the installer) - in other words, how does RAPPS know where to search for the uninstaller as well?